lib/Parse/Parser.cpp

```diff
@@ -907,6 +907,18 @@ bool Parser::parseMatchingToken(tok K, SourceLoc &TokLoc, Diag<> ErrorDiag,
   return false;
 }
 
+bool Parser::parseUnsignedInteger(unsigned &Result, SourceLoc &Loc,
```

Comment:
Why do you need this new method?  DAK_Alignment handling in ParseDecl.cpp does this:

   StringRef alignmentText = Tok.getText();
    unsigned alignmentValue;
    if (alignmentText.getAsInteger(0, alignmentValue)) {
      diagnose(Loc, diag::alignment_must_be_positive_integer);
      return false;
    }
    
Won't this work for you at the callsite?

Reply:
If you're trying to introduce a new generally useful utility, I'd recommend doing that as a separate patch which also migrates the existing callsites to it.

Reply:
I found the code snippet similar to `DAK_Alignment` in a bunch of places. I agree it is better to introduce this in a separate PR. I will do so shortly. 

There are also more uses of `parseUnsignedInteger` in the AD code base that has not been upstreamed yet.